src/mds/nameserver2/file_record.cpp

```diff
@@ -215,5 +215,15 @@ bool FileRecordManager::FindFileMountPoint(
     return true;
 }
 
+bool FileRecordManager::SureFileWriterLive(
+    const std::string& fileName, const std::string& ClientIP, uint32_t ClientPort) {
+    auto iter = fileRecords_[fileName];
+    butil::EndPoint ep;
+    int rc = butil::str2endpoint(clientIP.c_str(), ClientPort, &ep);
+    if(iter[ep].IsTimeout() == true) {
+        return true;
+    }
+    return false;
+}
```

Comment:
file record manager's state is volatile, it only keeps file records in memory, so if MDS failover, the new elected leader MDS doesn't have any file records.